samples/application_development/chre/README.rst

```diff
@@ -0,0 +1,47 @@
+Android's Context Hub Runtime Environment (CHRE)
+################################################
+
+Android's context hub enables the use of nanoapps. A single nanoapp has 3 entry points seen in
+`chre_api/chre/nanoapp.h`_:
+
+* A ``nanoappStart`` function used to notify the nanoapp that it is now active.
+* A ``nanoappHandleEvent`` function used to notify the nanoapp tha an event of interest took place.
+* A ``nanoappEnd`` function used to notify the nanoapp that it is now deactivated.
+
+The CHRE connects to several frameworks called Platform Abstraction Layers (PAL)s. Note that
+currently, none of these are implemented for Zephyr, but they are scheduled to be added. These
+frameworks include:
+
+#. *Audio* - a framework allowing nanoapps to get audio events. See `pal/audio.h`_ for API details.
+#. *GNSS* - a framework allowing nanoapps to manage location and measurement sessions. See
+   `pal/gnss.h`_ for API details.
+#. *Sensor* - a framework allowing nanoapps to request changes to sensors configuration get
+   data/bias events. See `pal/sensor.h`_ for API details.
+#. *System* - a framework allowing nanoapps to make common system requests such as logging, clock,
+   and some basic memory allocation/deallocation. See `pal/system.h`_ for API details.
+#. *WiFi* - a framework allowing nanoapps to interact with the on board WiFi. See `pal/wifi.h`_ for
+   API details.
+#. *WWAN* - a framework allowing nanoapps to interact with the WWAN module such as getting the
+   current capabilities and info. See `pal/wwan.h`_ for API details.
+
+Roadmap
+=======
+
+#. Add an implementation of the `pal/sensor.h`_ and `pal/system.h`_ to Zephyr. These will be
+   standalone modules that can be used independently of CHRE, but when ``CONFIG_CHRE`` is enabled
+   will also provide an implementation of ``chrePalSensorGetApi()`` and ``struct chrePalSystemApi``.
+#. Add a directory ``chre/nanoapps`` which will host various nanoapps to be used by the Zephyr
+   community. These should each have their own Kconfig to enable them and set the appropriate
+   dependencies. The first nanoapp will be a lid angle calculator which will use 2 accelerometers.
+#. Update the ``overlay.dts`` of this sample application to include 2 emulated accelerometers and
+   configure them to return scripted data.
+#. Run this sample application and watch the nanoapp provide lid angle calculations based on 2
+   accelerometers provided by the sensors PAL framework.
```

Comment:
Please add to the sample application README some instructions on how to build and a description of the expected output.

Reply:
Done